src/twisted/web/_flatten.py

```diff
@@ -229,50 +229,51 @@ def keepGoing(newRoot, dataEscaper=dataEscaper,
         write(b'-->')
     elif isinstance(root, Tag):
         slotData.append(root.slotData)
+        
         if root.render is not None:
             rendererName = root.render
             rootClone = root.clone(False)
             rootClone.render = None
             renderMethod = renderFactory.lookupRenderMethod(rendererName)
             result = renderMethod(request, rootClone)
             yield keepGoing(result)
-            slotData.pop()
-            return
-
-        if not root.tagName:
+        
```

Comment:
I don't what to say about these changes.

I would prefer to have `return` here, and maybe use something like `yield keepGoingFinal(result)` which will call slotData.pop()

With the current changes, slotData.pop() is called after many new branches... so this would need more than 1 new test.